Solution1/WeatherApi/Startup.cs

```diff
@@ -23,13 +28,22 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.Configure<AppConfiguration>(Configuration.GetSection(nameof(AppConfiguration)));
+            string connectionString = Configuration.GetConnectionString(Constants.Connection.NameDBConnection);
+            services.AddDbContextFactory<RepositoryContext>(options => options.UseSqlServer(connectionString));
+            
+            services.Configure<AppConfiguration>(Configuration.GetSection(Constants.ConfigurationSection.AppConfiguration));
+            services.Configure<PermanentRequestDTO>(Configuration.GetSection(Constants.ConfigurationSection.PermanentRequest));
             services.Configure<RouteOptions>(options => options.LowercaseUrls = true);
 
+            services.AddTransient<IStartupFilter, BackgroundJobFilter>();
```

Comment:
put to extension method with meaningful name: e.g. AddWeather...JobFilter

Reply:
+